model/task/task.go

```diff
@@ -3088,6 +3094,68 @@ func (t *Task) SetResetFailedWhenFinished() error {
 	)
 }
 
+func updateSchedulingLimitForResetWhenFinished(t *Task, caller string) error {
+	if !(t.Requester == evergreen.PatchVersionRequester || t.Requester == evergreen.GithubPRRequester) || evergreen.IsSystemActivator(caller) {
+		return nil
+	}
+	var tasks []Task
+	var err error
+	if t.DisplayOnly {
+		tasks, err = Find(ByIds(t.ExecutionTasks))
+		if err != nil {
+			return errors.Wrapf(err, "finding execution tasks for '%s'", t.Id)
+		}
+	} else if t.IsPartOfSingleHostTaskGroup() {
+		tasks, err = FindTaskGroupFromBuild(t.BuildId, t.TaskGroup)
+		if err != nil {
+			return errors.Wrapf(err, "finding task group '%s'", t.TaskGroup)
+		}
+	}
+	if len(tasks) == 0 {
+		return nil
+	}
+	return errors.Wrap(CheckUsersPatchTaskLimit(t.Requester, caller, tasks...), "updating patch task limit for user")
+}
+
+// CheckUsersPatchTaskLimit takes in an input list of tasks that is set to get activated, and checks if they're
+// non commit-queue patch tasks, and that the request has been submitted by a user. If so, the maximum hourly patch tasks counter
+// will be incremented accordingly.
+func CheckUsersPatchTaskLimit(requester, username string, tasks ...Task) error {
```

Comment:
This has been copied from [here](https://github.com/evergreen-ci/evergreen/pull/7891/files#diff-f50207584681d142f6e119549a8e2e22249b45dcbe3cb6f7570eb906be64299dL878-L906)